### PR TITLE
feat(components): Box Component

### PR DESCRIPTION
**Pull request by scotttjob**
## Motivations

We've had a need for a generic layout component for a while. This component was rolled elsewhere, and we're moving it into Atlantis so everyone can benefit from its greatness.

The Box component is for all your wrapping component needs. You can put borders around your box, or padding, or margins, or border radiuses. You can set flex direction, gap size, all sorts of fun things.

There is basic documentation, but could be expanded with additional examples.

## Changes

* Added the Box component

### Added

- The Box Component

## Testing

The box component should render well in the provided storybook stories.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Box/Box.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        radius && radiusStyles[`radius-${radius}`],
+        gap && gapStyles[`gap-${gap}`],
+      )}
+      style={{
+        ...getPaddingVars(padding),
+        ...getMarginVars(margin),
+        ...getHeightVars(height),
+        ...getWidthVars(width),
+        ...getBorderVars(border),
+        ...(background && { backgroundColor: `var(--color-${background})` }),
```
Comment (chris-at-jobber) [inline packages/components/src/Box/Box.tsx:62]: Is it possible to restrict the usages to semantic tokens? With the new `--color-base` prefix on the "base" tokens that might make it easier to do automatically, but basically ensuring that "if you use an Atlantis component, it responds to theme/mode overrides" is something we can continue to confidently promise

Comment (chris-at-jobber) [reply packages/components/src/Box/Box.tsx:62]: (Same note applies everywhere that we reference color in this file)

**Review by darryltec — COMMENTED** [review]
This LGTM. Tho @chris-at-jobber have a good question about background colours and it totally is possible, just need to write them manually now instead of inferring from all colour tokens

**Comment by chris-at-jobber** [comment]
In the name of getting this in the hands of people, I'm OK saying the semantic restrictions can be a fast-follow, but I'd hope it would be a _very_ fast follow as we have very semantic-token-reliant product work coming down the pipeline 😄

**Review by darryltec — APPROVED** [review]
In that case, ship it!

**Review by darryltec — APPROVED** [review]
go forth

**Pull request merged**